### PR TITLE
IGNITE-18638 Make tests print thread dump when timing out

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-18638

### CODE REVIEW

File: modules/core/src/testFixtures/resources/META-INF/services/org.junit.jupiter.api.extension.Extension [hunk 1]
```diff
@@ -0,0 +1 @@
+org.apache.ignite.internal.testframework.junit.DumpThreadsOnTimeout
```
Comment (sashapolo) [inline modules/core/src/testFixtures/resources/META-INF/services/org.junit.jupiter.api.extension.Extension:1]: Can we use auto-service instead?

File: modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/junit/DumpThreadsOnTimeout.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+ */
+public class DumpThreadsOnTimeout implements TestExecutionExceptionHandler, LifecycleMethodExecutionExceptionHandler {
+    @Override
+    public void handleTestExecutionException(ExtensionContext context, Throwable throwable) throws Throwable {
+        handleThrowable(throwable, "Test method");
+    }
+
+    private static void handleThrowable(Throwable throwable, String category) throws Throwable {
+        if (isJunitMethodTimeout(throwable)) {
+            System.out.println(category + " " + throwable.getMessage() + ", dumping threads");
```
Comment (sashapolo) [inline modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/junit/DumpThreadsOnTimeout.java:51]: Can you provide an example how this message will look like?

Comment (rpuch) [reply modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/junit/DumpThreadsOnTimeout.java:51]: https://ci.ignite.apache.org/buildConfiguration/ApacheIgnite3xGradle_Test_RunAllTests/7032302?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildTestsSection=true&expandBuildProblemsSection=true&expandCode+Inspection=true&pluginCoverage=true

It's just after the failure stack trace:

`Test method testIntLocking() timed out after 60 seconds, dumping threads`

File: modules/core/src/testFixtures/resources/META-INF/services/org.junit.jupiter.api.extension.Extension (hunk 1 above)
Comment (rpuch) [reply modules/core/src/testFixtures/resources/META-INF/services/org.junit.jupiter.api.extension.Extension]: Yes we can! Switched to auto-service.

**Review by sashapolo — APPROVED** [review]

**Pull request merged**